app/user-orders-repository.js

```diff
@@ -0,0 +1,34 @@
+const getUsersNames = async (db) => {
+    return new Promise((resolve, reject) => {
+        db.all(`SELECT DISTINCT name FROM USERS`, (err, rows) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(rows.map(row => row.name));
+            }
+        });
+    });
+};
+
+const getUserTotal = async (db, userName) => {
+    return new Promise((resolve, reject) => {
+        const sql = `SELECT SUM(total) as total FROM ORDERS o INNER JOIN USERS u ON u.id = o.user_id WHERE u.name = '${userName}'`;
+        db.all(sql, (err, rows) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve({ name: userName, total: rows[0].total });
+            }
+        });
+    });
+};
+
+module.exports.getMostValuableUsers = async (db, limit = 100) => {
```

Comment:
this code is inefficient and making too many DB calls.

all these functions together can be converted into a single one looking like this:
```
module.exports.getMostValuableUsers = async (db, limit = 100) => {
    return new Promise((resolve, reject) => {
        const sql = `
            SELECT u.name, SUM(o.total) as total
            FROM USERS u
            INNER JOIN ORDERS o ON u.id = o.user_id
            GROUP BY u.name
            ORDER BY total DESC
            LIMIT ?`;
        db.all(sql, [limit], (err, rows) => {
            if (err) {
                reject(err);
            } else {
                resolve(rows);
            }
        });
    });
};
```